src/registrar/views/portfolios.py

```diff
@@ -561,108 +557,42 @@ def get_success_url(self):
-                    # it has been sent but not accepted
-                    messages.warning(self.request, f"{email} has already been invited to this portfolio")
-                return
-        except Exception as err:
-            logger.error(f"_send_portfolio_invitation_email() => An error occured: {err}")
+            if not requested_user or not permission_exists:
+                send_portfolio_invitation_email(email=requested_email, requestor=requestor, portfolio=portfolio)
+                form.save()
+                messages.success(self.request, f"{requested_email} has been invited.")
+            else:
+                if permission_exists:
+                    messages.warning(self.request, "User is already a member of this portfolio.")
+        except Exception as e:
+            self._handle_exceptions(e, portfolio, requested_email)
+        return redirect(self.get_success_url())
 
-        try:
-            logger.debug("requestor email: " + requestor_email)
-
-            # send_templated_email(
-            #     "emails/portfolio_invitation.txt",
-            #     "emails/portfolio_invitation_subject.txt",
-            #     to_address=email,
-            #     context={
-            #         "portfolio": self.object,
-            #         "requestor_email": requestor_email,
-            #     },
-            # )
-        except EmailSendingError as exc:
-            logger.warn(
-                "Could not sent email invitation to %s for domain %s",
+    def _handle_exceptions(self, exception, portfolio, email):
+        """Handle exceptions raised during the process."""
+        if isinstance(exception, EmailSendingError):
+            logger.warning(
+                "Could not sent email invitation to %s for portfolio %s (EmailSendingError)",
                 email,
-                self.object,
-                exc_info=True,
-            )
-            raise EmailSendingError("Could not send email invitation.") from exc
-        else:
-            if add_success:
-                messages.success(self.request, f"{email} has been invited.")
-
-    def _make_invitation(self, email_address: str, requestor: User, add_success=True):
-        """Make a Member invitation for this email and redirect with a message."""
-        try:
-            self._send_portfolio_invitation_email(email=email_address, requestor=requestor, add_success=add_success)
-        except EmailSendingError:
-            logger.warn(
-                "Could not send email invitation (EmailSendingError)",
-                self.object,
+                portfolio,
                 exc_info=True,
             )
             messages.warning(self.request, "Could not send email invitation.")
-        except Exception:
-            logger.warn(
-                "Could not send email invitation (Other Exception)",
-                self.object,
+        elif isinstance(exception, MissingEmailError):
+            messages.error(self.request, str(exception))
```

Comment:
**suggestion:**  MissingEmailSendingError is initialized with a string that includes the username value, which looks like it would be the UUID for the user (correct me if I am wrong-I am skimming on a friday afternoon). Then here you are adding the exception with the str operator so that initialization string-including UUID- will show in the error message to the user. I normally wouldn't show a user a backend UUID if I can avoid it. I would suggest just overriding the __str__ function on the Missing email exception with a more user friendly error message

Reply:
Modified to a more friendly error message

Reply:
Thank you!